Sidekiqz2 build integration

commodo opened the pull request:
This changeset adds support for building SidekiqZ2 firmware images, similarly to PlutoSDR images.

The way it works:
* make - defaults to Pluto
* make TARGET=<target_name> - where <target_name> is `pluto` or `sidekiqz2`

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

rgetz commented:
I think that we also need some tweaks in legal_info_html.sh - so the written offer for sourcecode doesn't come from ADI...

I would just add a if [ "$TARGET" = "PlutoSDR" ] ; then around line 263.

and then an 
else
echo "Since you, the end user built this from source, for $TARGET, and didn't get a binary, there is no requirement for a written offer" >> ${FILE}
fi

around line 277. (above the </div>).

I can do that as a separate pull request if you want.

commodo commented:
Changelog v1 -> v2:
* updated .gitignore to ignore Vim swap files
* tweaked Legal Info based on comment https://github.com/analogdevicesinc/plutosdr-fw/pull/18#issuecomment-389922092  ( hopefully it's correct  now )
* parametrized Device VID and PID 
  * updated udev rules file `53-adi-plutosdr-usb.rules`
* re-organized device constants into own mk files [so pluto.mk & sidekiqz2.mk ]
* split commits a bit ; made commit `build: clean the build dir on each run` to make the intention clear about this change

commodo commented:
Changelog v2 -> v3:
* fix Legal generation for PlutoSDR (Written Offer) ; needed to pass the ADI_LEGAL constant to the `scripts/legal_info_html.sh` script as well

commodo commented:
Will merge this later if no objections.

mhennerich commented:
You need to close the box ...

diff --git a/scripts/legal_info_html.sh b/scripts/legal_info_html.sh
index 213dcdb..fcb9b4a 100755
--- a/scripts/legal_info_html.sh
+++ b/scripts/legal_info_html.sh
@@ -278,7 +278,7 @@ Since the source does not fit on a DVD-RW, it will be delivered on a USB Thumb d
 <p><b>You will also find a the source on-line, and are encouraged to obtain it for zero cost, at the project web sites.</b></p>
 </div>" >> ${FILE}
 else # not ADI_LEGAL
-echo "Since you, the end user built this from source, for ${TARGET}, and didn't get a binary, there is no requirement for a written offer." >> ${FILE}
+echo "Since you, the end user built this from source, for ${TARGET}, and didn't get a binary, there is no requirement for a written offer.</div>" >> ${FILE}
 fi
 
 html_h2 "NO WARRANTY"
diff --git a/u-boot-xlnx b/u-boot-xlnx

commodo commented:
ah ; 
will fix that on Monday

rgetz commented:
yeah, after you create the file, to test it, upload it to:

https://validator.w3.org/#validate_by_upload

it should pass without errors/warnings.
